src/DataStructures/Python/Tensor/TensorData.cpp

```diff
@@ -0,0 +1,50 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <memory>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/TensorData.hpp"
+#include "PythonBindings/BoundChecks.hpp"
+#include "Utilities/GetOutput.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+namespace py = pybind11;
+
+namespace py_bindings {
+void bind_tensordata(py::module& m) {  // NOLINT
+  // Wrapper for TensorComponent
+  py::class_<TensorComponent>(m, "TensorComponent")
+      .def(py::init<std::string, DataVector>(), py::arg("name"),
+           py::arg("data"))
+      .def_readwrite("name", &TensorComponent::name)
+      .def_readwrite("data", &TensorComponent::data)
+      .def("__str__", [](const TensorComponent& tc) { return get_output(tc); })
```

Comment:
Please do this instead (same for `__repr__`):

```cpp
.def("__str__", get_output<TensorComponent>)
```

Reply:
Out of curiosity, is this for readability or is there some performance benefit?

Reply:
I don't think the lambda function adds any significant performance overhead. But it's unnecessary, so we might as well remove it 🤷‍♂️

Reply:
I mostly just looked at the DataVector bindings for this so used the same format.